browser/src/map/Clipboard.js

```diff
@@ -887,45 +880,37 @@ L.Clipboard = L.Class.extend({
 		if (this._userAlreadyWarned('warnedAboutLargeCopy'))
 			return;
 
-		var self = this;
 		var msg = _('<p>If you would like to share larger elements of your document with other applications ' +
 			    'it is necessary to first download them onto your device. To do that press the ' +
 			    '"Start download" button below, and when complete click "Confirm copy to clipboard".</p>' +
 			    '<p>If you are copy and pasting between documents inside %productName, ' +
 			    'there is no need to download.</p>');
-		vex.dialog.alert({
-			unsafeMessage: this._substProductName(msg),
-			callback: function () {
-				self._map.focus();
-			}
-		});
+
+		this._map.uiManager.showInfoModal('large copy paste warning');
+		document.getElementById('large copy paste warning').innerHTML = this._substProductName(msg);
 	},
 
 	_warnLargeCopyPasteAlreadyStarted: function () {
-		var self = this;
-		vex.dialog.alert({
-			unsafeMessage: _('<p>A download due to a large copy/paste operation has already started. ' +
-				   'Please, wait for the current download or cancel it before starting a new one</p>'),
-			callback: function () {
-				self._map.focus();
-			}
-		});
+		this._map.uiManager.showInfoModal('large copy paste started warning');
+		document.getElementById('large copy paste started warning').innerHTML = _('<p>A download due to a large copy/paste operation has already started. ' +
+		'Please, wait for the current download or cancel it before starting a new one</p>');
+	},
+
+	isPasteSpecialDialogOpen: function() {
+		return document.getElementById('paste special dialog') ? true: false;
 	},
 
 	_openPasteSpecialPopup: function () {
-		var self = this;
 		var msg = _('<p>Your browser has very limited access to the clipboard</p><p>Please press now: <kbd>Ctrl</kbd><span class="kbd--plus">+</span><kbd>V</kbd> to see more options</p><p class="vex-footnote">Close popup to ignore paste special</p>');
 		msg = L.Util.replaceCtrlAltInMac(msg);
-		self._map._clip.pasteSpecialVex = vex.open({
-			unsafeContent: msg,
-			showCloseButton: true,
-			escapeButtonCloses: true,
-			overlayClosesOnClick: false,
-			buttons: {},
-			afterOpen: function() {
-				self._map.focus();
-			}
-		});
+
+		this._map.uiManager.showInfoModal('paste special dialog');
+
+		// showInfoModal creates the id in following format: 'modal-dialog-paste special dialog'.
+		// We will use this for closing the dialog.
+		this.pasteSpecialDialogId = 'modal-dialog-paste special dialog';
```

Comment:
It is fragile in case someone will change the modal dialog code generating id.
Maybe create helper function in UIManager: `generateModalId(id)` which will be reused in all the places where we typed: `var dialogId = 'modal-dialog-' + id;`. So there will be only one place to change in the future.

then this variable is not needed anymore. you will do `closeDialog(generateModalId('paste special dialog'));`